ocaml/typing/typemod.ml

```diff
@@ -3338,6 +3343,8 @@ let type_implementation sourcefile outputprefix modulename initial_env ast =
           signature = simple_sg
         } (* result is ignored by Compile.implementation *)
       end else begin
+        if !Clflags.as_parameter then
+          error Cannot_compile_implementation_as_parameter;
```

Comment:
I think currently if you pass `-i -as-parameter`, the `-as_parameter` is silently ignored? We should print some error along the line of "-i and -as-parameter" cannot be used together.

Reply:
This is mentioned in https://github.com/ocaml-flambda/flambda-backend/pull/1846#issuecomment-2192194104 and will be addressed there.